packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,163 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+ */
+export interface HttpGatewayRoutePathMatchConfig {
+  /**
+   * Gateway route configuration for matching on the complete URL path of the request.
+   *
+   * @default - no matching will be performed on the complete URL path
+   */
+  readonly wholePathMatch?: CfnGatewayRoute.HttpPathMatchProperty;
+
+  /**
+   * Gateway route configuration for matching on the prefix of the URL path of the request.
+   *
+   * @default - no matching will be performed on the prefix of the URL path
+   */
+  readonly prefixPathMatch?: string;
+
+  /**
+   * Gateway route configuration for rewriting the complete URL path of the request..
+   *
+   * @default - no rewrite will be performed on the request's complete URL path
+   */
+  readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * Gateway route configuration for rewriting the prefix of the URL path of the request.
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex: regex }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   When `prefixRewrite` is provided, `prefixPathMatch` must start and end with the '/' character.
+   *   If provided as "/", matches all requests.
+   *   For example, if your virtual service name is "my-service.local"
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param prefixRewrite Specify either disabling automatic rewrite to '/' or rewriting to specified prefix path.
+   */
+  public static startsWith(prefixPathMatch: string, prefixRewrite?: HttpGatewayRoutePathRewrite): HttpGatewayRoutePathMatch {
```

Comment:
1. This method should be first in this class, no?
2. The signature should be:

```suggestion
  public static startsWith(prefixPathMatch: string, rewriteTo: string | undefined = '/'): HttpGatewayRoutePathMatch {
```

Where passing `undefined` means disabling re-writing (`{ defaultPrefix: 'DISABLED' }`).
3. Explain in the docs that passing `null` / `undefined` means turning off re-writing.
4. You might want to consider treating `/` as a special value when it's passed for `rewriteTo`, and just leave the `rewrite` property as either `undefined`, or explicitly as `{ defaultPrefix: 'ENABLED' }` (your call).

Reply:
Please advise if I am understanding the logic correctly:

1. If `rewriteTo` is `null` / `undefined`, `rewriteTo` is assigned to `/`  
2. We are mandating an user to specify `rewriteTo` to enable the `defaultPrefix`
  - Wouldn't this be confusing for an user since our default setting is to have `defaultPrefix = enabled`

However, isn't scenario 1 equivalent to `default Prefix =enabled`? I must be missing something here.

Reply:
Should we have something like this instead?

```ts
public static startsWith(prefixPathMatch: string, rewriteTo: string | undefined): HttpGatewayRoutePathMatch {
    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, rewriteTo);
  }
```

```ts
class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
  constructor(
    private readonly prefixPathMatch: string,
    private readonly rewriteTo: string | undefined,
  ) { super(); }

  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
    if (this.rewriteTo) {
      return {
        prefixPathMatch: this.prefixPathMatch,
        prefixPathRewrite: {
          value: this.rewriteTo,
        },
      };
    }

    return {
      prefixPathMatch: this.prefixPathMatch,
      prefixPathRewrite: {
        defaultPrefix: 'DISABLED',
      },
    };
  }
}
```

Reply:
No. If `rewriteTo` is `undefined` (this can happen only if a customer specifies that explicitly), you set `{ defaultPrefix: 'DISABLED' }`. The default `rewriteTo` value is `/`, which, like I mentioned, you can handle in a special way in the L2 (by setting `{{ defaultPrefix: 'ENABLED' }`, if you want to, or leaving `rewrite` as `undefined`, and letting the service fill in the default).

Reply:
Changing the method signature to `public static startsWith(prefixPathMatch: string, rewriteTo: string = '/')`